### PR TITLE
Dev:make go test works on Mac

### PR DESCRIPTION
**Pull request by SamYuan1990**
By this pr, I try to make `go test ` working on Mac OS for development life easier.

Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

**Comment by rootfs** [comment]
@SamYuan1990 can you rebase?

File: Developer.md [hunk 1]
```diff
@@ -19,3 +19,13 @@ base on different k8s cluster, for example kind(k8s in docker)
 back to common checking process
 - deploy a kepler at local kind cluster with image stored at local image registry.
 - check via kubectl command for ...
+
+## Dev on Mac
+Please notice that in kepler we use different build tags to adopted with different hardwares.
```
Comment (rootfs) [inline Developer.md:24]: adapt to

**Comment by SamYuan1990** [comment]
@rootfs , could you please open an issue for `handle, _, err := unix.NameToHandleAt(unix.AT_FDCWD, path, 0)` the reason for split this file is due to `NameToHandleAt` is linux specific.

and my question is, for different between linux and mac, should we use build tag or runtime?
```
//go:build darwin
// +build darwin
```
or
```
if runtime.GOOS == "linux" {
```

**Comment by rootfs** [comment]
> @rootfs , could you please open an issue for `handle, _, err := unix.NameToHandleAt(unix.AT_FDCWD, path, 0)` the reason for split this file is due to `NameToHandleAt` is linux specific.
> 
> and my question is, for different between linux and mac, should we use build tag or runtime?
> 
> ```
> //go:build darwin
> // +build darwin
> ```
> 
> or
> 
> ```
> if runtime.GOOS == "linux" {
> ```

this could be done via a refactoring to split linux specific functions into a util pkg, do you like to give it a spin?

**Comment by SamYuan1990** [comment]
> > @rootfs , could you please open an issue for `handle, _, err := unix.NameToHandleAt(unix.AT_FDCWD, path, 0)` the reason for split this file is due to `NameToHandleAt` is linux specific.
> > and my question is, for different between linux and mac, should we use build tag or runtime?
> > ```
> > //go:build darwin
> > // +build darwin
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or
> > ```
> > if runtime.GOOS == "linux" {
> > ```
> 
> this could be done via a refactoring to split linux specific functions into a util pkg, do you like to give it a spin?

will try later.

**Comment by SamYuan1990** [comment]
I further checked with logic, and hope to further discussed at https://github.com/sustainable-computing-io/kepler/discussions/273 and for change related with #273 , will be in other prs.

File: .github/workflows/unit_test_mac.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Unit test Mac
+
+on:
+  pull_request:
+
+jobs:
+  unit_test_mac:
```
Comment (rootfs) [inline .github/workflows/unit_test_mac.yml:7]: should mac test be in GH action?

Comment (SamYuan1990) [reply .github/workflows/unit_test_mac.yml:7]:
```
    runs-on: macos-latest
```
yes? as the go test is going to running on masos based github action agent.

Comment (SamYuan1990) [reply .github/workflows/unit_test_mac.yml:7]: > should mac test be in GH action?

```
    runs-on: macos-latest
```
yes? as the go test is going to running on masos based github action agent.

Comment (SamYuan1990) [reply .github/workflows/unit_test_mac.yml:7]: In may point of view, I hope we can decouple kepler analysis(how to process data) with get data from infrastructure(how to get data).
hence I hope by this PR, we allow Mac user can help with how to process data.
For example, health check or rest api logic.

Comment (SamYuan1990) [reply .github/workflows/unit_test_mac.yml:7]: In may point of view, I hope we can decouple kepler analysis(how to process data) with get data from infrastructure(how to get data).
hence I hope by this PR, we allow Mac user can help with how to process data.
For example, health check or rest api logic.

**Review by rootfs — APPROVED** [review]

**Comment by rootfs** [comment]
let's see how the GH action works

**Pull request merged**